airflow/contrib/hooks/sqoop_hook.py

```diff
@@ -14,210 +14,282 @@
+            cmd += ["--append"]
 
-        cmd += ["--num-mappers", str(num_mappers)]
+        cmd += self._get_export_format_argument(file_type)
 
         if split_by:
             cmd += ["--split-by", split_by]
 
-        if append:
-            cmd += ["--append"]
+        if direct:
+            cmd += ["--direct"]
 
-        if type == "avro":
-            cmd += ["--as-avrodatafile"]
-        elif type == "sequence":
-            cmd += ["--as-sequencefile"]
-        else:
-            cmd += ["--as-textfile"]
+        if driver:
+            cmd += ["--driver", driver]
 
         return cmd
 
-    def import_table(self, table, target_dir,
-                     append=False, type="text", columns=None,
-                     num_mappers=None, split_by=None, where=None):
+    def import_table(self, table, target_dir, append=False, file_type="text",
+                     columns=None, split_by=None, where=None, direct=False,
+                     driver=None):
         """
         Imports table from remote location to target dir. Arguments are
         copies of direct sqoop command line arguments
         :param table: Table to read
         :param target_dir: HDFS destination dir
         :param append: Append data to an existing dataset in HDFS
-        :param type: "avro", "sequence", "text" Imports data to into the specified
-            format. Defaults to text.
+        :param file_type: "avro", "sequence", "text" or "parquet".
+            Imports data to into the specified format. Defaults to text.
         :param columns: <col,col,col…> Columns to import from table
-        :param num_mappers: Use n map tasks to import in parallel
         :param split_by: Column of the table used to split work units
         :param where: WHERE clause to use during import
+        :param direct: Use direct connector if exists for the database
+        :param driver: Manually specify JDBC driver class to use
         """
-        cmd = self._import_cmd(target_dir, append, type,
-                               num_mappers, split_by)
+        cmd = self._import_cmd(target_dir, append, file_type, split_by, direct,
+                               driver)
+
         cmd += ["--table", table]
+
         if columns:
             cmd += ["--columns", columns]
         if where:
             cmd += ["--where", where]
 
-        p = self.Popen(cmd, export=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        output, stderr = p.communicate()
+        self.Popen(cmd)
 
-        if p.returncode != 0:
-            # I like this better: RemoteCalledProcessError(p.returncode, cmd, self.host,
-            # output=output)
-            raise AirflowException("Cannot execute {} on {}. Error code is: "
-                                   "{}. Output: {}, Stderr: {}"
-                                   .format(cmd, self.conn.host,
-                                           p.returncode, output, stderr))
+    def import_query(self, query, target_dir,
```

Comment:
we should add options for extra args people may want to use. one extra args param for export and one for import is what I'd think would make sense.

Reply:
Sounds like a good idea